extensions/ql-vscode/src/skeleton-query-wizard.ts

```diff
@@ -0,0 +1,243 @@
+import { join } from "path";
+import { CancellationToken, Uri, workspace, window as Window } from "vscode";
+import { CodeQLCliServer } from "./cli";
+import { OutputChannelLogger } from "./common";
+import { Credentials } from "./common/authentication";
+import { QueryLanguage } from "./common/query-language";
+import { askForLanguage, isFolderAlreadyInWorkspace } from "./helpers";
+import { getErrorMessage } from "./pure/helpers-pure";
+import { QlPackGenerator } from "./qlpack-generator";
+import { DatabaseItem, DatabaseManager } from "./local-databases";
+import * as databaseFetcher from "./databaseFetcher";
+import { ProgressCallback } from "./progress";
+
+type QueryLanguagesToDatabaseMap = {
+  [id: string]: string;
+};
+
+export class SkeletonQueryWizard {
+  private language: string | undefined;
+  private folderName: string | undefined;
+  private fileName = "example.ql";
+  private storagePath: string | undefined;
+
+  QUERY_LANGUAGE_TO_DATABASE_REPO: QueryLanguagesToDatabaseMap = {
+    cpp: "protocolbuffers/protobuf",
+    csharp: "dotnet/efcore",
+    go: "evanw/esbuild",
+    java: "google/guava",
+    javascript: "facebook/react",
+    python: "pallets/flask",
+    ruby: "rails/rails",
+    swift: "Alamofire/Alamofire",
+  };
+
+  constructor(
+    private readonly cliServer: CodeQLCliServer,
+    private readonly progress: ProgressCallback,
+    private readonly credentials: Credentials | undefined,
+    private readonly extLogger: OutputChannelLogger,
+    private readonly databaseManager: DatabaseManager,
+    private readonly token: CancellationToken,
+  ) {
+    this.storagePath = this.workoutStoragePath();
+  }
+
+  public async execute() {
+    // show quick pick to choose language
+    await this.chooseLanguage();
+    if (!this.language) {
+      return;
+    }
+
+    this.folderName = `codeql-custom-queries-${this.language}`;
+    const skeletonPackAlreadyExists = isFolderAlreadyInWorkspace(
+      this.folderName,
+    );
+
+    if (skeletonPackAlreadyExists) {
+      // just create a new example query file in skeleton QL pack
+      await this.createExampleFile();
+      // select existing database for language
+      await this.selectExistingDatabase();
+    } else {
+      // generate a new skeleton QL pack with query file
+      await this.createQlPack();
```

Comment:
Thanks Shati. We want to treat the folder as a normal folder in the repo that you can commit and edit. 

So that means:
1. We can't save it in the extension storage, it needs to be in the workspace as a regular folder.
2. We should avoid confusing the user by also pushing the folder as a top level workspace folder (like you've discovered). 

So that means
- for the codespace repo: it needs to be created in the codespaces-codeql folder ✅ 
- for the starter repo: it needs to be a top level folder ✅ 
- we should open a follow-up PR to disable the "pushing as workspace folder" behaviour. 

The reason we were adding the folder to the workspace in the first place is because in the original flow we weren't interested in committing the folder so we went for storing it together with the database. That means the folder was invisible in the file explorer so we decided to make it visible by pushing it to the workspace as a top level folder. Hindsight is 20/20.